### PR TITLE
N°3767 - Impact analysis: Display filtering box on CIs list and groups

### PR DESCRIPTION
**Pull request by accognet**
The filter box is now above the tabs
image

image

### CODE REVIEW

**Comment by Molkobain** [comment]
Accepted during function review for 3.1.1
Technical review to do next.

Anne-Cath will have to rebase the branch **onto** support/3.1

**Comment by piRGoif** [comment]
Technical review : just Pierre & Anne Catherine attending. Pierre will do a review on its own.

File: pages/UI.php [hunk 1]
```diff
@@ -1410,64 +1410,66 @@ function DisplayNavigatorGroupTab($oP)
...
-				if (class_exists('Attachment')) {
-					$aAllowedClasses = MetaModel::GetModuleSetting('itop-attachments', 'allowed_classes', array('Ticket'));
-					foreach ($aAllowedClasses as $sAllowedClass) {
-						if ($oObj instanceof $sAllowedClass) {
-							$iIdForAttachment = $id;
-							$sClassForAttachment = $sClass;
-						}
+			$sClassIcon = MetaModel::GetClassIcon($sClass, false);
+			if (strlen($sClassIcon) > 0) {
+				$oTitle->SetIcon($sClassIcon);
```
Comment (piRGoif) [inline pages/UI.php]: The icon is now on the new title block, whereas it was on the class panel before. Can you explain with screenshots before/after why this modification ?

File: pages/UI.php [hunk 2]
```diff
@@ -1410,64 +1410,66 @@ function DisplayNavigatorGroupTab($oP)
...
-					DisplayNavigatorGroupTab($oP);
-				}
-				else
-				{
-					$oP->SetCurrentTab('UI:RelationshipGraph');
-					$oDisplayGraph->Display($oP, $aResults, $sRelation, $oAppContext, array(), $sClassForAttachment, $iIdForAttachment, $sContextKey, array('this' => $oObj),$bLazyLoading);
-					DisplayNavigatorListTab($oP, $aResults, $sRelation, $sDirection, $oObj);
-					DisplayNavigatorGroupTab($oP);
-				}
+			$oP->AddUiBlock($oTitle);
```
Comment (piRGoif) [inline pages/UI.php]: Can you move the $oTitle init here ? It's confusing to have block init and add so far away from each other :(

File: core/displayablegraph.class.inc.php [hunk 3]
```diff
@@ -1563,23 +1589,46 @@ function DoReload()
 	 * @throws \Twig\Error\LoaderError
 	 * @throws \Twig\Error\RuntimeError
 	 * @throws \Twig\Error\SyntaxError
+	 *
+	 * @deprecated 3.1.1
```
Comment (piRGoif) [inline core/displayablegraph.class.inc.php]: Tell why and what to use instead please

File: core/displayablegraph.class.inc.php [hunk 4]
```diff
@@ -1425,10 +1427,35 @@ public static function GetContextDefinitions($sContextKey, $bDevelopParams = tru
 	 *
 	 * @throws \CoreException
 	 * @throws \DictExceptionMissingString
+	 * @deprecated 3.1.1
```
Comment (piRGoif) [inline core/displayablegraph.class.inc.php]: Why deprecated ? This is still the method called in UI.php

File: pages/UI.php [hunk 5]
```diff
@@ -273,7 +274,6 @@ function DisplayNavigatorListTab($oP, $aResults, $sRelation, $sDirection, $oObj)
 	$oP->SetCurrentTab('UI:RelationshipList');
 	$oImpactedObject = UIContentBlockUIBlockFactory::MakeStandard("impacted_objects", ['ibo-is-visible']);
 	$oP->AddSubBlock($oImpactedObject);
-	$oImpactedObject->AddSubBlock(AlertUIBlockFactory::MakeForWarning(Dict::S("Relation:impacts/FilteredData"), '', "alert_filtered_list")->SetIsHidden(true));
```
Comment (piRGoif) [inline pages/UI.php:276]: I need an explanation on this removal. I see the dict key is used now in displayablegraph, but in a different type of block ?

File: core/displayablegraph.class.inc.php [hunk 6]
```diff
@@ -1626,11 +1675,23 @@ public function DisplayFiltering(string $sContextKey, array $aContextParams, arr
...
-		$oUiHtmlBlock->AddHtml("</div>\n"); // class="not-printable"
 
 		$oUiSearchBlock->AddSubBlock($oUiHtmlBlock);
-		$oP->AddUiBlock($oUiSearchBlock);
+		$oBlock->AddSubBlock($oUiSearchBlock);
 
+		return $oBlock;
+	}
+
+	public function GetFiltering(string $sContextKey, array $aContextParams, array $aExcludedObjects): array
```
Comment (piRGoif) [inline core/displayablegraph.class.inc.php]: To me the name is not very clear (filtering ?? uiblock, data, other ??). As far as I understand this is getting the filter options data ?

**Review by piRGoif — CHANGES_REQUESTED** [review]
To discuss with Anne Catherine in next review meeting

File: core/displayablegraph.class.inc.php (hunk 4 above)
Comment (piRGoif) [reply core/displayablegraph.class.inc.php]: Solved

File: core/displayablegraph.class.inc.php (hunk 3 above)
Comment (piRGoif) [reply core/displayablegraph.class.inc.php]: Don't forget this comment please O:)

File: pages/UI.php (hunk 1 above)
Comment (piRGoif) [reply pages/UI.php]: Discussed with Anne Catherine directly, closing conversation

File: pages/UI.php (hunk 2 above)
Comment (piRGoif) [reply pages/UI.php]: Don't forget this Anne Catherine !

File: pages/UI.php [hunk 7]
```diff
@@ -1406,64 +1406,66 @@ function DisplayNavigatorGroupTab($oP)
...
-				$oDisplayGraph = DisplayableGraph::FromRelationGraph($oRelGraph, $iGroupingThreshold, ($sDirection == 'down'));
-				$oPanel = PanelUIBlockFactory::MakeForClass($sClass, MetaModel::GetRelationDescription($sRelation, $bDirDown).' '.$oObj->GetName());
-				$sClassIcon = MetaModel::GetClassIcon($sClass, false);
-				if (strlen($sClassIcon) > 0){
-					$oPanel->SetIcon($sClassIcon);
-				}
+			$aResults = $oRelGraph->GetObjectsByClass();
+			$oDisplayGraph = DisplayableGraph::FromRelationGraph($oRelGraph, $iGroupingThreshold, ($sDirection == 'down'));
+			$sIconUIrl = MetaModel::GetClassIcon($sClass, false);
+			$oTitle = TitleUIBlockFactory::MakeForPage(MetaModel::GetRelationDescription($sRelation, $bDirDown).' '.$oObj->GetName());
```
Comment (piRGoif) [inline pages/UI.php]: To check with @steffunky or @Molkobain if there is a better looking component for this ?

File: pages/UI.php (hunk 5 above)
Comment (accognet) [reply pages/UI.php:276]: Filter box is now always visible. This warning message is therefore unnecessary.

**Comment by piRGoif** [comment]
The PR was reviewed and accepted in last functional review
Just the oTitle potential replacement to be checked, and it'll be ok to merge !

**Comment by Molkobain** [comment]
I still have to review it as well as we discussed it with Anne-Catherine a while back.

**Review by Molkobain — APPROVED** [review]

**Pull request merged**